### ISSUE TITLE
Different results between TSDF on GPU and CPU when using  noisy data

### ISSUE THREAD

**Issue by HM102** [intel-isl/Open3D#3332]
I am using the same parameters for both CPU and GPU TSDF as below:

```
volume_cpu = open3d::pipelines::integration::ScalableTSDFVolume(8.0 / 512.0, 0.04,open3d::pipelines::integration::TSDFVolumeColorType::RGB8);
 
volume_gpu = open3d::t::geometry::TSDFVoxelGrid({{"tsdf", open3d::core::Dtype::Float32},
                                           {"weight", open3d::core::Dtype::UInt16},
                                           {"color", open3d::core::Dtype::UInt16}},
                                          8.f / 512.f,  0.04f, 16,
                                          1000, device);
```

For the CPU version I extract the point cloud;
```
auto pcd_cpu = volume_cpu->ExtractPointCloud();
open3d::io::WritePointCloudToPLY("pointcloud_cpu.ply",*pcd_cpu,true);
```
![image](https://user-images.githubusercontent.com/43437535/115887372-2762ec00-a452-11eb-9967-fec639f54243.png)

For the GPU, first I used the default one with threshold=3.0
```
auto pcd_gpu = volume_gpu->ExtractSurfacePoints();
auto pcd_gpu_legacy = std::make_shared<open3d::geometry::PointCloud>(pcd_gpu.ToLegacyPointCloud());
```
![image](https://user-images.githubusercontent.com/43437535/115887416-32b61780-a452-11eb-9a18-026eca844d96.png)

and threshold =0.0, which according to #2927  I should get the same results as the CPU version, since all frames are included:
`auto pcd_gpu_2 = volume_gpu->ExtractSurfacePoints(0.0);`

![image](https://user-images.githubusercontent.com/43437535/115887448-3b0e5280-a452-11eb-8dd6-956c276f453d.png)

I get an improved result, however it still not the same as the CPU version!

**Comment by theNded** [comment]
First, it is not CPU vs CUDA, but legacy vs CUDA. The pipelines are different, so there could be other parameters changed. It would be helpful to see the rest of the code.

Then, for the data how many frames are you fusing? And which sequence in TUM are you using? What is the open3d version?

**Comment by HM102** [comment]
Hi @theNded thanks for the replay. 

I am using the freiburg1_room  sequence with around 200 frames ( subset of all frames ~1350). Open3D version 0.12.


```
auto trajectory =  io::CreatePinholeCameraTrajectoryFromFile(trajectory_path);
camera::PinholeCameraIntrinsic intrinsic = camera::PinholeCameraIntrinsic( camera::PinholeCameraIntrinsicParameters::PrimeSenseDefault);

auto focal_length = intrinsic.GetFocalLength();
auto principal_point = intrinsic.GetPrincipalPoint();
Tensor intrinsic_t = Tensor(std::vector<float>({static_cast<float>(focal_length.first), 0,static_cast<float>(principal_point.first), 0, static_cast<float>(focal_length.second),
static_cast<float>(principal_point.second), 0,0, 1}), {3, 3}, Dtype::Float32);
```
```

for (size_t i = 0; i < trajectory->parameters_.size() - 1; ++i) {
  
      std::shared_ptr<geometry::Image> depth_legacy = io::CreateImageFromFile(depth_filenames[i]);
      
      std::shared_ptr<geometry::Image> color_legacy =io::CreateImageFromFile(color_filenames[i]);
      
      Eigen::Matrix4d extrinsic = trajectory->parameters_[i].extrinsic_;
      Tensor extrinsic_t = core::eigen_converter::EigenMatrixToTensor(extrinsic).To(device);
      std::shared_ptr<geometry::RGBDImage> rgbd = geometry::RGBDImage::CreateFromColorAndDepth(*color_legacy, *depth_legacy, 5000.0, 4.0, false);
      
      volume_cpu.Integrate(*rgbd, intrinsic, extrinsic);
       
      t::geometry::Image depth_cuda = t::geometry::Image::FromLegacyImage(rgbd->depth_, device);
      t::geometry::Image color_cuda = t::geometry::Image::FromLegacyImage(rgbd->color_, device);
      t::geometry::RGBDImage rgbd_t = open3d::t::geometry::RGBDImage(color_cuda,depth_cuda);
      
      volume_gpu.Integrate(rgbd_t.depth_, rgbd_t.color_, intrinsic_t, extrinsic_t, 1.0, 4.0);

}
```

**Comment by theNded** [comment]
![image](https://user-images.githubusercontent.com/6127282/116437079-0cf68d00-a81b-11eb-9b18-9f65444f55a6.png)
This is what I get for the first 200 frames.

**Comment by HM102** [comment]
I fixed the problem, it was caused by the conversion done similar to #3260. The conversion worked for the legacy, however for the GPU it did not.

When I read the depth images directly as 16 bit images with a depth scale of 5000 it works.

GPU
![image](https://user-images.githubusercontent.com/43437535/116545335-8d3af200-a8f0-11eb-802e-6e3672c677c8.png)

Legacy
![image](https://user-images.githubusercontent.com/43437535/116545470-c1aeae00-a8f0-11eb-99f0-3ed73dd3ab0a.png)
